contrib/builder/dockerfiles/generate.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+set -e
+
+# usage: ./generate.sh [versions]
+#    ie: ./generate.sh
+#        to update all Dockerfiles in this directory
+#    or: ./generate.sh debian-jessie
+#        to only update debian-jessie/Dockerfile
+#    or: ./generate.sh debian-newversion
+#        to create a new folder and a Dockerfile within it
+
+cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+
+types=( arm gccgo ppc64le s390x )
```

Comment:
totally ya that would be most ideal

On Tue, Dec 8, 2015 at 2:17 PM, Tianon Gravi <notifications@github.com>
wrote:

> In contrib/builder/dockerfiles/generate.sh
> <https://github.com/docker/docker/pull/18415#discussion_r47022504>:
>
> > @@ -0,0 +1,156 @@
> > +#!/bin/bash
> > +set -e
> > +
> > +# usage: ./generate.sh [versions]
> > +#    ie: ./generate.sh
> > +#        to update all Dockerfiles in this directory
> > +#    or: ./generate.sh debian-jessie
> > +#        to only update debian-jessie/Dockerfile
> > +#    or: ./generate.sh debian-newversion
> > +#        to create a new folder and a Dockerfile within it
> > +
> > +cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
> > +
> > +types=( arm gccgo ppc64le s390x )
>
> I think we either need to also auto-generate the main Dockerfile, or pull
> sections from it dynamically here so that we don't still have two files
> that need to be in sync (on top of needing to re-run the generate script
> whenever we update).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18415/files#r47022504>.
>